src/common/snippets/src/lowered/loop_manager.cpp

```diff
@@ -438,6 +464,25 @@ void LinearIR::LoopManager::fuse_loops(LinearIR::constExprIt loop_begin_target,
     remove_loop_info(from);
 }
 
+std::vector<lowered::pass::PassPipeline> LinearIR::LoopManager::fuse_loop_handlers(
+    const std::vector<lowered::pass::PassPipeline>& lhs,
+    const std::vector<lowered::pass::PassPipeline>& rhs) {
+    OPENVINO_ASSERT(lhs.size() == rhs.size(), "fuse_loop_handlers supports only handlers vectors with equal sizes.");
+    auto merged_handlers = lhs;
+    for (size_t i = 0; i < lhs.size(); ++i) {
+        const auto& res_passes = merged_handlers[i].get_passes();
+        for (const auto& pass : rhs[i].get_passes()) {
+            auto pred = [&pass](const std::shared_ptr<lowered::pass::PassBase>& p) {
+                return p->get_type_info() == pass->get_type_info();
```

Comment:
I introduced a virtual method `bool can_be_merged(const std::shared_ptr<PassBase>& other)` in a `PassBase` class. This method defines if 2 passes can be merged during pipelines fusion or not. The method returns `false` by default.
If some pass is used in loop iter handlers, we need to override the method.  In classes like `SetLoadCount` with some members (e.g. `count`) we usually compare `count` value and return true if counts are equal.
So if we try to merge `SetLoadCount(16)` and `SetLoadCount(32)`, exception will be thrown since these 2 class instances can't be merged into one.